runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeItemPanel.java

```diff
@@ -0,0 +1,127 @@
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.grandexchange;
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Desktop;
+import java.awt.GridLayout;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.text.NumberFormat;
+import javax.swing.ImageIcon;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.border.CompoundBorder;
+import javax.swing.border.EmptyBorder;
+import javax.swing.border.LineBorder;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+class GrandExchangeItemPanel extends JPanel
+{
+	GrandExchangeItemPanel(BufferedImage icon, String name, int itemID, int gePrice, Double haPrice)
+	{
+		BorderLayout layout = new BorderLayout();
+		layout.setHgap(5);
+		setLayout(layout);
+		setToolTipText(name);
+
+		addMouseListener(new MouseAdapter()
+		{
+			@Override
+			public void mouseClicked(MouseEvent e)
+			{
+				geLink(name, itemID);
+			}
+		});
+
+		setBorder(new CompoundBorder
+		(
+			new LineBorder(getBackground().brighter(), 1),
+			new EmptyBorder(5, 5, 5, 5)
+		));
+
+		// Icon
+		JLabel itemIcon = new JLabel();
+		itemIcon.setIcon(new ImageIcon(icon));
+		add(itemIcon, BorderLayout.LINE_START);
+
+		// Item details panel
+		JPanel rightPanel = new JPanel(new GridLayout(3, 1));
+
+		// Item name
+		JLabel itemName = new JLabel();
+		itemName.setText(name);
+		rightPanel.add(itemName);
+
+		// Ge price
+		JLabel gePriceLabel = new JLabel();
+		gePriceLabel.setText(NumberFormat.getInstance().format(gePrice) + " gp");
+		gePriceLabel.setForeground(Color.GREEN);
+		rightPanel.add(gePriceLabel);
+
+		// Alch price
+		JLabel haPriceLabel = new JLabel();
+		haPriceLabel.setText(NumberFormat.getInstance().format(haPrice.intValue()) + " alch");
+		haPriceLabel.setForeground(Color.orange);
+		rightPanel.add(haPriceLabel);
+
+		add(rightPanel, BorderLayout.CENTER);
+	}
+
+	public void geLink(String name, int itemID)
+	{
+		String url = "http://services.runescape.com/m=itemdb_oldschool/" + name.replaceAll(" ", "_") + "/viewitem?obj=" + itemID;
+
+		if (!Desktop.isDesktopSupported())
+		{
+			log.info("Desktop is not supported. Visit {}", url);
+			return;
+		}
+
+		Desktop desktop = Desktop.getDesktop();
+		if (!desktop.isSupported(Desktop.Action.BROWSE))
+		{
+			log.info("Desktop browser is not supported. Visit {}", url);
+			return;
+		}
+
+		try
+		{
+			desktop.browse(new URI(url));
+
+			log.debug("Opened browser to {}", url);
+		}
+		catch (IOException | URISyntaxException ex)
+		{
+			log.warn("Unable to open grand exchange page", ex);
```

Comment:
We should really start popping up message boxes in cases like this. See #434 . Maybe create service that will handle opening the links (and showing message box with the link in case of opening of the link is unsupported) to not repeat #434 and to not duplicate code every time someone would like to open a link somewhere.